hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/meta/TestCkpMetadata.java

```diff
@@ -49,11 +52,21 @@ public class TestCkpMetadata {
   @TempDir
   File tempFile;
 
+  protected Configuration conf;
+
+  protected HoodieFlinkWriteClient writeClient;
+
   @BeforeEach
   public void beforeEach() throws Exception {
+    setup();
+  }
+
+  protected void setup() throws IOException {
     String basePath = tempFile.getAbsolutePath();
-    Configuration conf = TestConfigurations.getDefaultConf(basePath);
+    this.conf = TestConfigurations.getDefaultConf(basePath);
+    conf.setString(HoodieWriteConfig.INSTANT_STATE_TIMELINE_SERVER_BASED.key(), "false");
     StreamerUtil.initTableIfNotExists(conf);
+    this.writeClient = FlinkWriteClients.createWriteClient(conf);
   }
```

Comment:
Not sure why we instantiate a server write client here, did you know the code would start an embedded timeline server there? Actually the coordinator would also start an embedded timeline server, where all the requests should be rootted too.

Reply:
The tests may workout but you need to be cautious here, the CkpMetadata has read/write methods, the read methods should always be invoked from the client, while the write methods should be from driver, it is greate if you can imitate with the reality cases.